djangocms_version_locking/monkeypatch/admin.py

```diff
@@ -1,46 +1,183 @@
+
+    # Check version exists
+    version = self.get_object(request, unquote(object_id))
+    if version is None:
+        return self._get_obj_does_not_exist_redirect(
+            request, self.model._meta, object_id)
+
+    # Raise 404 if not locked
+    if version.state != constants.DRAFT:
+        raise Http404
+
+    # Check that the user has unlock permission
+    if not request.user.has_perm('djangocms_version_locking.delete_versionlock'):
+        return HttpResponseForbidden(force_text(_("You do not have permission to remove the version lock")))
+
+    # Unlock the version
+    remove_version_lock(version)
+    # Display message
+    messages.success(request, _("Version unlocked"))
+
+    # Redirect
+    url = admin_reverse('{app}_{model}_changelist'.format(
+        app=self.model._meta.app_label,
+        model=self.model._meta.model_name,
+    )) + '?grouper=' + str(version.grouper.pk)
+
+    return redirect(url)
+admin.VersionAdmin._unlock_view = _unlock_view
+
+
+def _get_unlock_link(self, obj, request):
+    """
+    Generate an unlock link for the Versioning Admin
+    """
+    # If the version is not draft no action should be present
+    if obj.state != constants.DRAFT:
+        return ""
+
+    # Check whether the lock can be removed
+    # Check that the user has unlock permission
+    if is_draft_locked(obj) and request.user.has_perm('djangocms_version_locking.delete_versionlock'):
+        unlock_url = reverse('admin:{app}_{model}_unlock'.format(
+            app=obj._meta.app_label, model=self.model._meta.model_name,
+        ), args=(obj.pk,))
+
+        return render_to_string(
+            'djangocms_version_locking/admin/unlock_icon.html',
+            {'unlock_url': unlock_url}
+        )
+
+    return render_to_string(
+        'djangocms_version_locking/admin/unlock_disabled_icon.html',
+    )
+admin.VersionAdmin._get_unlock_link = _get_unlock_link
+
+
+def _get_urls(func):
+    """
+    Add custom Version Lock urls to Versioning urls
+    """
+    def inner(self, *args, **kwargs):
+        url_list = func(self, *args, **kwargs)
+        info = self.model._meta.app_label, self.model._meta.model_name
+        url_list.insert(0,
+            url(
+                r'^(.+)/unlock/$',
+                self.admin_site.admin_view(self._unlock_view),
+                name='{}_{}_unlock'.format(*info),
+            )
+        )
+        return url_list
+    return inner
+admin.VersionAdmin.get_urls = _get_urls(admin.VersionAdmin.get_urls)
+
+
+def get_state_actions(func):
+    """
+    Add custom Version Lock actions to Versioning state actions
+    """
+    def inner(self, *args, **kwargs):
+        state_list = func(self, *args, **kwargs)
+        state_list.append(self._get_unlock_link)
+        return state_list
+    return inner
+admin.VersionAdmin.get_state_actions = get_state_actions(admin.VersionAdmin.get_state_actions)
+
+
+def edit_redirect_view(func):
```

Comment:
Patch `_get_edit_redirect_version` instead by calling the original function first, checking if version is None and then updating the author